search/django/weedid/views.py

```diff
@@ -91,6 +95,74 @@ def upload(request):
         )
 
 
+def upload_voc(request):
+    if not request.method == "POST":
+        return HttpResponseNotAllowed(request.method)
+    user = request.user
+    if not (user and user.is_authenticated):
+        return HttpResponseForbidden("You dont have access to proceed")
+    try:
+        voc_dir = request.POST["voc_dir"]
+        voc = request.FILES["voc"]
+        upload_dir = os.path.join(UPLOAD_DIR, str(user.id), voc_dir)
+        store_tmp_voc(voc, upload_dir)
+    except Exception as e:
+        return HttpResponseBadRequest(str(e))
+    else:
+        return HttpResponse(f"Uploaded {voc.name} to {upload_dir}")
+
+
+def remove_voc(request):
+    if not request.method == "POST":
+        return HttpResponseNotAllowed(request.method)
+    user = request.user
+    if not (user and user.is_authenticated):
+        return HttpResponseForbidden("You dont have access to proceed")
+    voc_dir = request.POST["voc_dir"]
+    voc_name = request.POST["voc_name"]
+    voc_to_remove = os.path.join(UPLOAD_DIR, str(user.id), voc_dir, voc_name)
+    if os.path.exists(voc_to_remove):
+        os.remove(voc_to_remove)
+        return HttpResponse(f"Removed {voc_name}")
+    else:
```

Comment:
I'd do this with try-except. Apart from anything else, it avoids race conditions. (This code could raise an exception if two identical remove_voc requests were issued at the same time.)